src/app/containers/BlockChain/Governance/Staking/saga.ts

```diff
@@ -0,0 +1,165 @@
+// import { take, call, put, select, takeLatest } from 'redux-saga/effects';
+// import { actions } from './slice';
+
+import { parseEther } from "ethers/lib/utils";
+import { call, put, select, takeLatest } from "redux-saga/effects";
+import { selectMainTokenABIDomain } from "app/containers/BlockChain/selectors";
+import { StakingActions } from "./slice";
+import { CreateLockData, DistributorData } from "./types";
+import { ethers } from "ethers";
+import { env } from "environment";
+import { getEpochSecondForDay } from "./helpers/date";
+import { selectGovernanceTokenContract } from "../selectors";
+import { BlockChainActions } from "../../slice";
+import { toast } from "react-toastify";
+import {
+  selectAccountDomain,
+  selectLibraryDomain,
+} from "app/containers/BlockChain/Web3/selectors";
+import {
+  selectFeeDistributorABIDomain,
+  selectOtherDistributorsDomain,
+} from "./selectors";
+
+export function* createLock(action: { type: string; payload: CreateLockData }) {
+  const { balance, date } = action.payload;
+  const amount = parseEther(balance.toString());
+  yield put(StakingActions.setIsStaking(true));
+  const library = yield select(selectLibraryDomain);
+  //|| is used because if .env is not set,we will fetch the error in early stages
+  const mainTokenAddress = env.MAIN_TOKEN_ADDRESS || "";
+  const mainTokenABI = yield select(selectMainTokenABIDomain);
+  try {
+    const mainTokenContract = new ethers.Contract(
+      mainTokenAddress,
+      mainTokenABI,
+      library.getSigner()
+    );
+    if (mainTokenContract) {
```

Comment:
consider converting `ifelse` into a switch, when possible. Or abstract out functionality into a helper function.
The Idea is to have small function bodies and avoid cyclomatic complexity.